Improve description of fuzzy matching in editor settings documentation

Meorge opened the pull request:
Similar to #107108, this PR clarifies what "fuzzy matching" does in the documentation for the Enable Fuzzy Matching editor setting.

Meorge committed:
Improve description of fuzzy matching in editor settings documentation

Update doc/classes/EditorSettings.xml

Co-authored-by: Micky <66727710+Mickeon@users.noreply.github.com>

Update `max_fuzzy_misses` docs and "Fuzzy Search" tooltip to match `enable_fuzzy_matching`

Improve examples used and remove "middle of the filename" description

Apply suggestions from code review

Co-authored-by: Micky <66727710+Mickeon@users.noreply.github.com>
Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

Update doc/classes/EditorSettings.xml

Co-authored-by: Rémi Verschelde <rverschelde@gmail.com>

Improve phrasing for `enable_fuzzy_matching` description

Update doc/classes/EditorSettings.xml

Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

Semicolons

Meorge commented on doc/classes/EditorSettings.xml:
```diff
@@ -784,7 +784,8 @@
 		<member name="filesystem/quick_open_dialog/enable_fuzzy_matching" type="bool" setter="" getter="">
-			If [code]true[/code], fuzzy matching of search tokens is allowed.
+			If [code]true[/code], include inexact matches or matches starting in the middle of a filename, instead of only exact matches from the beginning. For example, with fuzzy matching enabled, searching [code]foobar[/code] may find [code]foobur[/code] despite not being a perfect match.
+			Use [member filesystem/quick_open_dialog/max_fuzzy_misses] to set how much the search query can differ from the match results.
```
Sounds good, will work on that!

Mickeon commented:
> The focus for this description should be on the fuzzy matching behavior, so frontloading the description of the non-fuzzy behavior instead may be confusing or make it take a bit longer for readers to reach the info they're interested in.

I suppose I agree with that. Right now, I am not familiar with fuzzy matching enough to say what _exactly_ needs to be said.

Mickeon reviewed (commented):
In hindsight, "addons" would be more familiar than "addon" as that's the name of the... addons folder.
But now that I think about it, Quick Opening a folder is impossible. So I am not sure what to suggest. `icon`?

I just tested, though. If I disable fuzzy matching with the Quick Open Dialog's toggle, "matches starting in the middle" are included anyway. Are you sure they are handled by fuzzy matching?

Meorge commented:
For now, I changed it to "normal" since that's a pretty typical word, but also one that someone might reasonably expect to have in individual filenames. On my end, with fuzzy search off the searches still matched mid-filename, so I removed that from the description here.

Mickeon reviewed (commented):
Almost done, amen!

Meorge commented on doc/classes/EditorSettings.xml:
```diff
@@ -784,13 +784,14 @@
 		</member>
 		<member name="filesystem/quick_open_dialog/enable_fuzzy_matching" type="bool" setter="" getter="">
-			If [code]true[/code], fuzzy matching of search tokens is allowed.
+			If [code]true[/code], together with exact matches of a filename, the dialog includes approximate matches. For example, with fuzzy matching enabled, searching [code]"normal"[/code] may find [code]"narmal"[/code] despite not being a perfect match.
```
"narmal" as a typo of "normal" is what I had in mind, but I agree that "nprmal" is a much better, more realistic/likely example.

akien-mga replied:
Yeah, though my point is also that the examples were in the wrong order IMO, as said that searching "normal" would find "narmal", when it should be the other way around to be realistic/useful.

AThousandShips commented on doc/classes/EditorSettings.xml:
```diff
@@ -784,13 +784,15 @@
 		<member name="filesystem/quick_open_dialog/enable_fuzzy_matching" type="bool" setter="" getter="">
-			If [code]true[/code], fuzzy matching of search tokens is allowed.
+			If [code]true[/code], together with exact matches of a filename, the dialog includes approximate matches.
+			This is useful for finding the correct files even when there are typos in the search query, e.g., searching "nprmal" will find "normal". Additionally, it allows you to write shorter search queries, e.g., searching "nml" will also find "normal".
```
```suggestion
			This is useful for finding the correct files even when there are typos in the search query, for example, searching "nprmal" will find "normal". Additionally, it allows you to write shorter search queries, for example, searching "nml" will also find "normal".
```
Consistency with abbreviations

Meorge replied:
Sounds good to me, although I think the preceding commas will need to be replaced with semicolons to avoid being run-on sentences
